src/util/fields.py

```diff
@@ -1079,8 +1078,23 @@ def _update_field_usage_multi(cr, models, old, new, domain_adapter=None, skip_in
         """
         cr.execute(q.format(col_prefix=col_prefix), p)
 
-        # ir.exports.line
+        # ir.exports.line, base_import.mapping # noqa
         if only_models:
+
+            def fix_paths(potential_broken_paths, old_field_name, new_field_name):
```

Comment:
Note: now that we do this more than once, you can move all one in this if block to a `_update_impex_fields_paths` to be in line with the other new function ;)

Untested! just a copy-pasta example:
```py
def _update_impex_fields_paths(cr, model, old, new):
    export_q = cr.mogrify(
        """
        SELECT el.id,
               e.resource,
               STRING_TO_ARRAY(el.name, '/')
          FROM ir_exports_line el
          JOIN ir_exports e
            ON el.export_id = e.id
         WHERE el.name ~ %s
        """,
        [r"\y{}\y".format(old)],
    ).decode()
    impex_data = [("ir_exports_line", "name", export_q)]
    if table_exists(cr, "base_import_mapping"):
        import_q = cr.mogrivy(
            """
            SELECT id,
                   res_model,
                   STRING_TO_ARRAY(field_name, '/')
              FROM base_import_mapping
             WHERE field_name ~ %s
            """,
            [r"\y{}\y".format(old)],
        ).decode()
        impex_data.append(("base_import_mapping", "field_name", import_q))
    
    for table, column, query in impex_data:
        cr.execute(query)
        if not cr.rowcount:
            continue
        fixed_paths = {}
        for record_id, related_model, path in cr.fetachall():
            new_path = [
                new
                if field.field_name == old and field.field_model in only_models
                else field.field_name
                for field in resolve_model_fields_path(cr, related_model, path)
            ]
            if len(new_path) == len(path) and new_path != path:
                fixed_paths[record_id] = "/".join(new_path)
        if fixed_paths:
            cr.execute(
                format_query(cr, "UPDATE {} SET {} = (%s::jsonb)->>(id::text) WHERE id IN %s", table, column),
                [Json(fixed_paths), tuple(fixed_paths)],
            )
```

Reply:
You can keep it inline, but this entry function `rename_field` is getting very long already, since we have a similar function for removal (although it's true that the latter is used in multiple places) we could have it extracted into a new funciton with a relevant name and the code here would look neater.